bin/elixir.ps1

```diff
@@ -0,0 +1,379 @@
+See run_erl to learn more. To reattach, run: to_erl PIPEDIR.
+
+--pipe-to is not supported on Windows. If set, Elixir won't boot.
+
+** Options marked with (*) can be given more than once.
+** Standalone options can't be combined with other options.
+"@
+
+  Write-Host $Help
+}
+
+if (($Args.Count -eq 1) -and ($Args[0] -eq "--short-version")) {
+  Write-Host "$ELIXIR_VERSION"
+  exit
+}
+
+if (($Args.Count -eq 0) -or (($Args.Count -eq 1) -and ($Args[0] -in @("-h", "--help")))) {
+  PrintElixirHelp
+  exit 1
+}
+
+function NormalizeArg {
+  param(
+    [Parameter(Mandatory = $true, ValueFromPipeline = $true)]
+    [string[]] $Items
+  )
+  [string]::Join(",", $Items)
+}
+
+function QuotedString {
+  param(
+    [Parameter(Mandatory = $true, ValueFromPipeline = $true)]
+    [string] $Item
+  )
+
+  # We surround the string with double quotes, in order to preserve its contents as
+  # only one command arg.
+  # This is needed because PowerShell will consider a new argument when it sees a
+  # space char.
+  if (-not $Item.StartsWith("`"") -and $Item.Contains(" ")) {
+    "`"$Item`""
+  }
+  else {
+    $Item
+  }
+}
+
+$ElixirParams = New-Object Collections.Generic.List[String]
+$ErlangParams = New-Object Collections.Generic.List[String]
+$BeforeExtras = New-Object Collections.Generic.List[String]
+$AllOtherParams = New-Object Collections.Generic.List[String]
+
+$RunErlPipe = $null
+$RunErlLog = $null
+
+for ($i = 0; $i -lt $Args.Count; $i++) {
+  $private:Arg = $Args[$i]
+
+  switch ($Arg) {
+    { $_ -in @("-e", "-r", "-pr", "-pa", "-pz", "--eval", "--remsh", "--dot-iex", "--dbg") } {
+      $private:NextArg = $Args[++$i] | NormalizeArg | QuotedString
+
+      $ElixirParams.Add($Arg)
+      $ElixirParams.Add($NextArg)
+
+      break
+    }
+
+    { $_ -in @("-v", "--version", "--help", "-h") } {
+      # Standalone options goes only once in the Elixir params, when they are empty.
+      if ($ElixirParams.Count -eq 0) {
+        $ElixirParams.Add($Arg)
+      }
+      else {
+        $AllOtherParams.Add($Arg)
+      }
+      break
+    }
+
+    "--no-halt" {
+      $ElixirParams.Add($Arg)
+      break
+    }
+
+    "--cookie" {
+      $ErlangParams.Add("-setcookie")
+      $ErlangParams.Add($Args[++$i])
+      break
+    }
+
+    "--hidden" {
+      $ErlangParams.Add("-hidden")
+      break
+    }
+
+    "--name" {
+      $ErlangParams.Add("-name")
+      $ErlangParams.Add($Args[++$i])
+      break
+    }
+
+    "--sname" {
+      $ErlangParams.Add("-sname")
+      $ErlangParams.Add($Args[++$i])
+      break
+    }
+
+    "--boot" {
+      $ErlangParams.Add("-boot")
+      $ErlangParams.Add($Args[++$i])
+      break
+    }
+
+    "--erl-config" {
+      $ErlangParams.Add("-config")
+      $ErlangParams.Add($Args[++$i])
+      break
+    }
+
+    "--vm-args" {
+      $ErlangParams.Add("-args_file")
+      $ErlangParams.Add($Args[++$i])
+      break
+    }
+
+    "--logger-otp-reports" {
+      $private:TempVal = $Args[$i + 1]
```

Comment:
Style has changed from `++$i` to `$i + 1`.  I think the latter is kinder to the reader.

Reply:
I see.  I'd not seen `++$i` in PowerShell before so had to look it up.